lib/seattleflu/id3c/cli/command/etl/fhir.py

```diff
@@ -50,6 +51,53 @@ def create_patient_resource(patient_identifier: List[dict], gender: str) -> dict
     })
 
 
+def create_diagnostic_report(redcap_record:dict,
+                             patient_reference: dict,
+                             specimen_reference: dict,
+                             create_device_result_observation_resource: Callable) -> Optional[dict]:
+    """
+    Create FHIR diagnostic report from given *redcap_record*. Attaches
+    observation resources to it using the given, device-specific
+    *create_device_result_observation_resource* function.
+
+    Links the generated diagnostic report to a specific *patient_reference* and
+    *specimen_reference*.
+    """
+    clinical_results = create_device_result_observation_resource(redcap_record)
+    if not clinical_results:
+        return None
+
+    diagnostic_result_references = []
+    for result in clinical_results:
+        reference = create_reference(
+            reference_type = 'Observation',
+            reference = '#' + result['id']
+        )
+        diagnostic_result_references.append(reference)
+
+    collection_datetime = redcap_record['collection_date']
+
+    diagnostic_code = create_codeable_concept(
+        system = 'http://loinc.org',
+        code = '85476-0',
+        display = 'FLUAV and FLUBV and RSV pnl NAA+probe (Upper resp)'
+    )
```

Comment:
This diagnostic code is specifice to the Cepheid assay.

Reply:
I see. Do you know how to find the right code for these tests?

Reply:
I think this might work for the clinical report: https://loinc.org/77029-7/
It seems general enought to encompass all the different pathogens we ingest.

Reply:
Or this one https://loinc.org/60566-7/ is even more general

Reply:
Thank you!

Reply:
This is another one https://loinc.org/78922-2/
There's so many I don't even know which one fits the best here....
(I'm just googling "loinc respiratory pathogen")

Reply:
There are so many... I think I will go with the most general ("Unspecified") one for now. I'm not sure how to choose the appropriate one.

Reply:
Found the code on [UW](https://testguide.labmed.uw.edu/public/view/REVSQT) under `Billing/Coding`. It's https://loinc.org/50219-5/

Reply:
> Found the code on [UW](https://testguide.labmed.uw.edu/public/view/REVSQT) under `Billing/Coding`. It's https://loinc.org/50219-5/

Awesome find! I used the [UW Laboratory Test Guide](https://testguide.labmed.uw.edu/public/index) website you linked to look up the codes for all the tests.

Reply:
Ughh. Another FHIR rabbit hole.
We can only have one `DiagnosticReport.code` value per resource. However, it is possible to have multiple entries for a `codeableConcept`. From the [FHIR docs](http://hl7.org/fhir/datatypes.html#CodeableConcept): 
> More than one code may be used in CodeableConcept. The concept may be coded multiple times in different code systems (or even multiple times in the same code systems, where multiple forms are possible, such as with SNOMED CT). 

Right now we have three LOINC codes for a DiagnosticReport: 
```
    uw_billing_to_loinc_map = {
        'NCVQLT': '94306-8',
        'REVSQT': '50219-5',
        'FLUMC': '48509-4',
    }
```

I'm not sure if it's appropriate to include all three LOINC codes in the `DiagnosticReport.code.codeableConcept` list, because they're all mutually exclusive. I think the true-to-FHIR way would be to insert a new DiagnosticReport for each LOINC code. Thoughts?

Reply:
@tsibley @joverlee521 any thoughts here?

Reply:
I agree that the true FHIR way is a new DiagnosticReport for each diagnostic code. I want to say that the FHIR ETL is capable of handling multiple DiagnosticReports in one Bundle.